evaluator/src/test/kotlin/ProjectSourceRuleTest.kt

```diff
@@ -62,6 +62,38 @@ class ProjectSourceRuleTest : WordSpec({
             rule.projectSourceHasFile("README.md").matches() shouldBe false
         }
     }
+
+    "projectSourceHasDirectory()" should {
+        "return true if at least one directory matches the given glob pattern" {
+            val dir = createSpecTempDir().apply {
+                addDirs("a/b/c")
+            }
+            val rule = createOrtResultRule(dir)
+
+            with(rule) {
+                projectSourceHasDirectory("a").matches() shouldBe true
+                projectSourceHasDirectory("a/b").matches() shouldBe true
+                projectSourceHasDirectory("**/b/**").matches() shouldBe true
+                projectSourceHasDirectory("**/c").matches() shouldBe true
+            }
+        }
+
+        "return false if only a file matches the given glob pattern" {
+            val dir = createSpecTempDir().apply {
+                addFiles("a")
+            }
+            val rule = createOrtResultRule(dir)
+
+            rule.projectSourceHasDirectory("a").matches() shouldBe false
+        }
+
+        "return false if neither any file nor directory matches the given glob pattern" {
```

Comment:
The test should either create files and directories that do not match or be renamed to "return false if the directory is empty".
Currently you have no test that verifies that the pattern does not match any path.